src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -61,26 +92,46 @@ public VisionMode getMode() {
         } catch(Exception e) {
             // ignore
         }
-        return VisionMode.INTAKE;
+        return DEFAULT_MODE;
     }
 
     public boolean getStatus() {
-        double[] visionData = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
         // 1 = success, but it comes as a float, so test with a greater-than
-        return visionData[1] > 0.5;
+        return m_lastResult[1] > 0.5;
     }
     
     public double getDistance() {
-        double[] visionData = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
-        return visionData[3];
+        return m_lastSuccessfulResult[3];
     }
     
     public double getRobotAngle() {
-        double[] visionData = SmartDashboard.getNumberArray("vision/target_info", EMPTY_TARGET_INFO);
-        return Math.toDegrees(visionData[4]);
+        return Math.toDegrees(m_lastSuccessfulResult[4]);
     }
 
     public void setLedRing (boolean on) {
         m_spike.set(on ? Value.kReverse : Value.kOff);
     }
+
+    public class AwaitVisionResult extends CommandBase {
```

Comment:
This should be a separate command in with all the other commands. Also you can't reset the mode at the end, otherwise the Shooter will not get a distance.

Reply:
It is only reset if it is interrupted (because in a sequential command group the command that's running is responsible for cleaning up)